### ISSUE TITLE
Unifont HEX fonts

### ISSUE THREAD

**Issue by polyzium** [fnuecke/oc2#210]
OC2 is currently using an atlas of Terminus as its default font, however I'd think it's necessary to have the support for Unifont HEX format fonts, just like in OC1 (which uses unscii16 I think?), which is very beneficial for Unicode support and compatibility with applications expecting an OC1 display, for a possible opportunity of OC1 emulation inside OC2's RISC-V engine ala payonel's ocvm.

**Comment by SilverSCode** [comment]
It would be nice to actually support Unicode character sets

**Comment by SilverSCode** [comment]
Does this make it possible to display multiple languages?

**Comment by polyzium** [comment]
yes it does. Check the original OpenComputers mod and its config.

**Comment by fnuecke** [comment]
I'm inclined to keep terminal ASCII only for now, since the projector could be used for arbitrary charset rendering.

**Comment by polyzium** [comment]
The projector has a resolution of 640x480 which limits the terminal resolution to 80x30. The built in terminal is around the same number of chars (80x25 I presume?), and if an OC1 emulator is to be developed, it would be impossible to use Tier3 screen/GPU requiring programs and graphical applications, among other things.

**Comment by atirutw** [comment]
> if an OC1 emulator is to be developed, it would be impossible to use Tier3 screen/GPU requiring programs and graphical applications, among other things.

And I'm assuming you have to go away from the main computer(the only input source) to even look at the projection?

**Comment by polyzium** [comment]
@atirut-w The projector has its own keyboard for input and has its own tty, and the built-in terminal has its own input and tty as well.

**Comment by atirutw** [comment]
I see

**Comment by asiekierka** [comment]
I think hardware font glyph drawing acceleration could still be cool, for two reasons:

* font size - the .hex file currently used by OpenComputers is over 4 megabytes (now that Unifont is under a license which makes bundling it with OC1 less concerning). In a binary format, it could probably be dropped to around 2 megabytes - however, that's a lot of data to be stored on every computer wishing to render Unicode text,
* bandwidth - ... I had an entire text in mind about accelerated draw operations, but you're just using H.264? wh-

**Comment by polyzium** [comment]
Oh hello there, @asiekierka.

I'm afraid the kernel would have to be modified for this. The projector is seen by Linux as a framebuffer device and would treat it as such, rendering the whole glyphs pixel by pixel all by itself without any help from the mod's rendering facilities. But I'm also afraid that would break the framebuffer support for displaying images. However it would be nice to have Unicode support in computer's built-in terminal at least, as it is much simpler to implement.

1. That could require making a brand new format for font data. I'd say stick to gold standards such as PCF or BDF formats. We don't want another useless format taking place and fragmenting the "standards" world.
2. Judging by the drawing speed I think it's just raw bitmaps. I don't think it would be updated much, something like MJPEG would do the job just fine in this case.

~~p.s. pls update computronics to work with this kthx bye~~

**Comment by asiekierka** [comment]
Small errata: Computronics is @Vexatos, not me. Also, H.264 seems to be what the projector is using right now, it's not a proposal!

**Comment by polyzium** [comment]
Projector is using a proper video codec? Then why is it so slow? It should be this slow when transferring raw bitmap data, especially when the FB is 640x480@R5G6B5.

**Comment by fnuecke** [comment]
When it doubt, it's slow because it's throttled, both network bandwidth and the speed of the virtual cpu. The default terminal will remain as-is, won't learn unicode I think. Projector should already be able to.
